tempesta_fw/cache.c

```diff
@@ -457,7 +457,7 @@ __cache_add_node(TDB *db, TfwHttpResp *resp, TfwHttpReq *req,
 	size_t data_len = sizeof(*ce) + req->uri_path.len
 			  + req->h_tbl->tbl[TFW_HTTP_HDR_HOST].len
 			  + sizeof(TfwCStr) * resp->h_tbl->off
-			  + resp->msg.len;
+			  + resp->body.len;
```

Comment:
Hm, why? ```data_len``` is the length of data stored in cache as TfwCacheEntry. Plus to ```ce```, descriptor of the stored cache entry, we store URI and Host header - this is cache key which is used to match stored cache entry against request. Next we store binary representation of all response headers to be able to quickly read them and generate a response ready for possible headers adjustments. ```TfwCStr``` is used to describe each header string. So so far we calculated only length of auxilary data. So we add length of whole response message including status line, all the header and message body - all the things will be stored.